stan/math/prim/fun/vec_concat.hpp

```diff
@@ -7,31 +7,55 @@
 
 namespace stan {
 namespace math {
-
+namespace internal {
+inline auto sum_vector_sizes() { return 0; }
+/**
+ * Get the internal sizes of a pack of vectors.
+ * @tparam Vec Type of first vector to count.
+ * @tparam VecArgs Parameter pack of vector types to accumulate sizes of
+ * @param x vector to start accumulation of sizes with.
+ * @param args pack of vectors to accumulate sizes for.
+ */
+template <typename Vec, typename... VecArgs>
+inline auto sum_vector_sizes(Vec&& x, VecArgs&&... args) {
+  return x.size() + sum_vector_sizes(std::forward<VecArgs>(args)...);
```

Comment:
```suggestion
inline auto sum_vector_sizes(const Vec& x, const VecArgs&... args) {
  return x.size() + sum_vector_sizes(args...);
```

No need for forwarding here.